crates/oxc_linter/src/rule.rs

```diff
@@ -117,15 +130,56 @@ where
 
         let value = serde_json::Value::deserialize(deserializer)?;
 
-        if let serde_json::Value::Array(arr) = value {
-            let config = arr
-                .into_iter()
-                .next()
-                .and_then(|v| serde_json::from_value(v).ok())
-                .unwrap_or_else(T::default);
-            Ok(DefaultRuleConfig(config))
-        } else {
-            Err(D::Error::custom("Expected array for rule configuration"))
+        // Even if it only has a single type parameter T, we still expect an array
+        // for ESLint-style rule configurations.
+        let serde_json::Value::Array(arr) = value else {
+            return Err(D::Error::custom("Expected array for rule configuration"));
+        };
+
+        // Empty array -> use T::default() (fast-path; no clones)
+        if arr.is_empty() {
+            return Ok(DefaultRuleConfig(T::default()));
+        }
+
+        // Match on the array contents:
+        // - [arg] where arg is an object => try parsing `arg` as T; if that
+        //   fails, attempt to parse [arg, {}] into T (helps tuple-of-objects)
+        //   otherwise fall back to T::default().
+        // - otherwise => try parsing the whole array as T (tuple form). If
+        //   that fails, parse the first element into T or fall back to default.
+        match arr.as_slice() {
+            [first] if first.is_object() => {
+                // Try the simple object-as-T path first.
+                if let Ok(config) = serde_json::from_value::<T>(first.clone()) {
+                    return Ok(DefaultRuleConfig(config));
+                }
```

Comment:
is there any way we can avoid the clones here? When i originally wrote this I was super careful about the clones, as this is a hot-ish path, so avoiding them where we can is ideal.